packages/react-storage/src/components/StorageImage/StorageImage.tsx

```diff
@@ -1,53 +1,109 @@
 import * as React from 'react';
 
-import { classNames, ComponentClassName, isUndefined } from '@aws-amplify/ui';
+import { classNames, ComponentClassName } from '@aws-amplify/ui';
 import { Image } from '@aws-amplify/ui-react';
-import { useStorageURL } from '@aws-amplify/ui-react/internal';
-import { useSetUserAgent } from '@aws-amplify/ui-react-core';
+import { useDeprecationWarning } from '@aws-amplify/ui-react/internal';
+import { useGetUrl, useSetUserAgent } from '@aws-amplify/ui-react-core';
 
 import { VERSION } from '../../version';
-import type { StorageImageProps } from './types';
+import type { StorageImageProps, StorageImagePathProps } from './types';
 
-export const StorageImage = ({
+export const StorageImageWithKey = ({
   accessLevel,
   className,
   fallbackSrc,
   identityId,
   imgKey,
+  path,
   onStorageGetError,
-  validateObjectExistence,
+  validateObjectExistence = true,
   ...rest
 }: StorageImageProps): JSX.Element => {
-  const resolvedValidateObjectExistence = isUndefined(validateObjectExistence)
-    ? true
-    : validateObjectExistence;
-  const options = React.useMemo(
-    () => ({
-      accessLevel,
-      targetIdentityId: identityId,
-      validateObjectExistence: resolvedValidateObjectExistence,
-    }),
-    [accessLevel, identityId, resolvedValidateObjectExistence]
-  );
-
-  useSetUserAgent({
-    componentName: 'StorageImage',
-    packageName: 'react-storage',
-    version: VERSION,
+  if (imgKey && path) {
+    throw new Error('StorageImage cannot have both imgKey and path props.');
+  }
+  useDeprecationWarning({
+    message:
+      'The `imgKey` prop has been deprecated and will be removed in the next major version of Amplify UI.',
+    shouldWarn: true,
```

Comment:
we should only show it when `imgKey` prop is in use?

Reply:
Yes, did a little bit of refactoring but it will only show the warning if using `imgKey`